### PR TITLE
refactor(rpc_address): simple refactor on rpc_address

### PR DESCRIPTION
**Pull request by acelyc111**
https://github.com/apache/incubator-pegasus/issues/1234

Simple refactor before add feature to support FQDN. Including:
1. move rpc_address.* from `src/utils/` to `src/runtime/rpc/`
2. move `src/utils/test/address.cpp` to `src/runtime/test/address_test.cpp`
3. add warning log for `rpc_group_address.add/remove` callers, and will mark TODO to add `WARN_UNUSED_RESULT`
4. simple refactor for `group_address.h`, `rpc_address.h|cpp`
5. minimize the include header for `rpc_address.h`

### CODE REVIEW

File: src/runtime/rpc/rpc_address.cpp [hunk 1]
```diff
@@ -51,15 +54,15 @@ uint32_t rpc_address::ipv4_from_host(const char *name)
...
     if ((addr.sin_addr.s_addr = inet_addr(name)) == (unsigned int)(-1)) {
+        // TODO(yingchun): use getaddrinfo instead
         hostent *hp = ::gethostbyname(name);
-        int err = h_errno;
-
-        if (hp == nullptr) {
-            LOG_ERROR("gethostbyname failed, name = %s, err = %d.", name, err);
+        if (dsn_unlikely(hp == nullptr)) {
+            LOG_ERROR_F(
+                "gethostbyname failed, name = {}, err = {}", name, utils::safe_strerror(h_errno));
```
Comment (empiredan) [inline src/runtime/rpc/rpc_address.cpp:61]: Human-readable message for `::gethostbyname` can be got by `hstrerror(h_errno)`, however considering that soon we will use `getaddrinfo`, we can use `gai_strerror` instead then.

File: src/runtime/rpc/rpc_address.h [hunk 2]
```diff
@@ -151,10 +134,15 @@ class rpc_address
     // and you MUST ensure that _addr is INITIALIZED before you call this function
     void set_invalid();
 
-    bool operator==(::dsn::rpc_address r) const
+    bool operator==(rpc_address r) const
```
Comment (empiredan) [inline src/runtime/rpc/rpc_address.h:137]:
```suggestion
    bool operator==(const rpc_address &r) const
```

File: src/runtime/rpc/rpc_address.h [hunk 3]
```diff
@@ -166,9 +154,9 @@ class rpc_address
         }
     }
 
-    bool operator!=(::dsn::rpc_address r) const { return !(*this == r); }
+    bool operator!=(rpc_address r) const { return !(*this == r); }
```
Comment (empiredan) [inline src/runtime/rpc/rpc_address.h:157]:
```suggestion
    bool operator!=(const rpc_address &r) const { return !(*this == r); }
```

File: src/runtime/rpc/rpc_address.h [hunk 4]
```diff
@@ -166,9 +154,9 @@ class rpc_address
...
 
-    bool operator<(::dsn::rpc_address r) const
+    bool operator<(rpc_address r) const
```
Comment (empiredan) [inline src/runtime/rpc/rpc_address.h:159]:
```suggestion
    bool operator<(const rpc_address &r) const
```

File: src/utils/utils.h [hunk 5]
```diff
@@ -28,8 +28,10 @@
 
 #include <functional>
 #include <memory>
+#include <map>
```
Comment (empiredan) [inline src/utils/utils.h:31]:
```suggestion
#include <map>
#include <memory>
```

**Review by empiredan — APPROVED** [review]

**Review by levy5307 — APPROVED** [review]

**Pull request merged**